src/Waives.Pipelines/HttpAdapters/HttpDocument.cs

```diff
@@ -37,9 +37,19 @@ public async Task<ClassificationResult> Classify(string classifierName)
 
         public async Task Delete(Action afterDeletedAction)
         {
-            await _documentClient.Delete().ConfigureAwait(false);
+            try
+            {
+                await _documentClient.Delete().ConfigureAwait(false);
+            }
+            catch (Exception e)
```

Comment:
:muscle: Catch block is unnecessary.
In fact, the `afterDeletedAction` param is also unnecessary - calling code can do the following:
`doc.Delete().ContinueWith(() => {})` instead.

Reply:
Ha, removing that action was one of the changes I had on my branch - it's always called with the no-op action now.